api/pkg/apis/v1alpha1/managers/solution/solution-manager.go

```diff
@@ -61,6 +62,7 @@ type SolutionManager struct {
 	StateProvider   states.IStateProvider
 	ConfigProvider  config.IExtConfigProvider
 	SecretProvider  secret.ISecretProvider
+	KeyLockProvider keylock.IKeyLockProvider
```

Comment:
Let's also remove global lock in solution-manager to avoid any mis-use in future.

Reply:
removed